src/main/java/ui/UI.java

```diff
@@ -143,13 +143,29 @@ private void login(boolean isBypassLogin) {
         getMainWindowHandle(mainStage.getTitle());
     }
 
+    private void openSampleBoard(){
+        BoardAutoCreator boardCreator = new BoardAutoCreator(this, panels, prefs);
+        boardCreator.createSampleBoard(false);
+    }
+
     private void disableUI(boolean disable) {
         mainStage.setResizable(!disable);
         menuBar.setDisable(disable);
         repoSelector.setDisable(disable);
     }
 
+    /**
+     * Returns true if HubTurbo is not being run on Test mode or if startup board creation is being tested.
+     * As for the other tests, they start on a clean state with dummy repos established
+     * so don't need boards created on startup.
+     */
+    private boolean shouldOpenSampleBoard() {
+        return !TestController.isTestMode() || TestController.shouldTestStartupBoard();
+    }
+
     private void showMainWindow(String repoId) {
+        boolean isFirstTimeUser = logic.getStoredRepos().isEmpty();
```

Comment:
This is the closest to where it can be. This has to be here since `logic.getStoredRepos()` gets altered in the next line by `logic.openPrimaryRepository`.